### DISCUSSION TITLE
Possible memory leak using @react-navigation/native-stack in react-native-web (reproducible)

### DISCUSSION THREAD

**Discussion by hoangvvo** [react-navigation/react-navigation#10458] [Q&A]
I am suspecting a memory leak in react-navigation, specifically in the `headerBack` component. But I am not sure this is the case so instead of a bug report I created a discussion.

I noticed that every time I navigate to a different native-stack page and back, the amount of Node keep increases. My expectation is that it should get GC after I navigate back.  On Android, after multiple navigation like this, the web app would just **freeze**.
![image](https://user-images.githubusercontent.com/40987398/160234460-9a5c3bcd-0eca-4905-ab78-59f3b10e0c69.png)

So I took a snapshot using devtool and found a single detached `SVGSVGElement` consumes lots of memory compared to others. And inside itself seems to what appears to be self-references. This SVG is used in `headerLeft` for the native-stack.

![image](https://user-images.githubusercontent.com/40987398/160234477-b84b9d46-3af9-4ef2-affa-3af7dc839f9e.png)
![image](https://user-images.githubusercontent.com/40987398/160234480-5053af1a-2fd9-49e2-a121-17a2a4fe4010.png)

I thought it is something wrong with my back component, so I remove it to use react-navigation default one, but seeing similar occurrence. 
![image](https://user-images.githubusercontent.com/40987398/160234514-96515ee1-fe17-4fac-9ba4-cf2f9118ba26.png)

The HTMLDivElement corresponds to `<div aria-label="Go back" role="button" tabindex="0" class="css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-touchAction-1otgn73" style="margin: 3px 11px; min-width: 1px; opacity: 0.30056;">...</div>`, the default `headerBack`, so this happens even without my component.

I can also replicate this using the example app https://react-navigation-example.netlify.app/native-stack

https://user-images.githubusercontent.com/40987398/160235174-c2c84e2b-1252-4fd4-8692-fef0cf2cc7ea.mp4

**Comment by hoangvvo** [comment]
Related to #7078 and #7686

**Comment by satya164** [comment]
Thanks for the investigation. I don't have much time to investigate this currently but it seems pretty weird that the button is still in memory even after unmounting. As far as I remember, we're not keeping a reference to it anywhere. Let me know if you manage to find the cause.
